### PR TITLE
[Conv2D] Apply padding props to conv2d

### PR DESCRIPTION
**Pull request by zhoonit**
~**[Pending commit: #1357]**~

- [Conv2D] Apply padding props to conv2d

```
This patch applies padding property to conv2d

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/layers/conv2d_layer.h [hunk 1]
```diff
@@ -126,7 +127,8 @@ class Conv2DLayer : public LayerV1 {
   unsigned int filter_size;
   std::array<unsigned int, CONV2D_DIM> kernel_size;
   std::array<unsigned int, CONV2D_DIM> stride;
-  std::array<unsigned int, CONV2D_DIM> padding;
+  std::array<unsigned int, 4> padding;
```
Comment (kparichay) [inline nntrainer/layers/conv2d_layer.h:130]: `std::array<unsigned int, CONV2D_DIM * 2> padding;`?

File: nntrainer/layers/conv2d_layer.cpp [hunk 2]
```diff
@@ -66,16 +68,16 @@ static void col2im(const Tensor &col_matrix, const TensorDim &kdim,
   unsigned wdilation = dilation[1];
 
   /// image considering padding
-  unsigned im_eff_height = im_height + 2 * ph;
-  unsigned im_eff_width = im_width + 2 * pw;
+  unsigned im_eff_height = im_height + pt + pb;
```
Comment (kparichay) [inline nntrainer/layers/conv2d_layer.cpp:71]: i recommend to make inline function for these.

File: nntrainer/layers/conv2d_layer.cpp [hunk 3]
```diff
@@ -170,14 +172,16 @@ static void im2col(const Tensor &in, const TensorDim &kdim,
...
+  unsigned pb = padding[1];
+  unsigned pl = padding[2];
+  unsigned pr = padding[3];
 
   unsigned int channel = in.channel();
   int in_height = in.height();
   int in_width = in.width();
-  unsigned int height = in_height + ph * 2;
-  unsigned int width = in_width + pw * 2;
+  unsigned int height = in_height + pt + pb;
```
Comment (kparichay) [inline nntrainer/layers/conv2d_layer.cpp:183]: inline func for this as well.

File: nntrainer/layers/conv2d_layer.cpp [hunk 4]
```diff
@@ -66,16 +68,16 @@ static void col2im(const Tensor &col_matrix, const TensorDim &kdim,
...
   /// image considering padding
-  unsigned im_eff_height = im_height + 2 * ph;
-  unsigned im_eff_width = im_width + 2 * pw;
+  unsigned im_eff_height = im_height + pt + pb;
+  unsigned im_eff_width = im_width + pl + pr;
   image.setZero();
 
-  int h_stride_end = im_eff_height - eff_k_height - ph;
-  int w_stride_end = im_eff_width - eff_k_width - pw;
+  int h_stride_end = im_eff_height - eff_k_height - pb;
```
Comment (kparichay) [inline nntrainer/layers/conv2d_layer.cpp:75]: inline func for this as well.

**Review by kparichay — APPROVED** [review]
overall LGTM other than some minor comments

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/layers/conv2d_layer.cpp (hunk 2 above)
Comment (zhoonit) [reply nntrainer/layers/conv2d_layer.cpp:71]: I have a little clean up idea for this things, let me delay this until v2 is merged :)

File: nntrainer/layers/conv2d_layer.h (hunk 1 above)
Comment (zhoonit) [reply nntrainer/layers/conv2d_layer.h:130]: Done

File: nntrainer/layers/conv2d_layer.cpp (hunk 2 above)
Comment (kparichay) [reply nntrainer/layers/conv2d_layer.cpp:71]: cool.

**Pull request merged**